cishouseholds/pipeline/high_level_transformations.py

```diff
@@ -2529,7 +2533,197 @@ def union_dependent_cleaning(df):
         covered_enclosed_column="face_covering_other_enclosed_places",
         covered_work_column="face_covering_work_or_education",
     )
-
+    think_had_covid_cols = [
+        "survey_response_dataset_major_version",
+        "think_had_covid_admitted_to_hospital",
+        "think_had_covid",
+        "think_had_covid_contacted_nhs",
+        "sympt_covid_count",
+        "think_had_covid_onset_date",
+    ]
+    hospital_covid_cols = ["think_had_covid_contacted_nhs", "think_had_covid_admitted_to_hospital"]
+    other_covid_test_cols = ["other_covid_infection_test", "other_covid_infection_test_results"]
+    covid_test_cols = [*hospital_covid_cols, *other_covid_test_cols]
+    # 1
+    df = assign_column_value_from_multiple_column_map(
+        df,
+        "think_had_covid_admitted_to_hospital",
+        [
+            None,
+            [1, "No", "No", "No", 0, None],
+        ],
+        think_had_covid_cols,
+    )
+    # 2
+    df = assign_column_value_from_multiple_column_map(
+        df,
+        "think_had_covid_contacted_nhs",
+        [
+            None,
+            [1, None, "No", "No", 0, None],
+        ],
+        think_had_covid_cols,
+    )
+    # 3
+    for col in ["think_had_covid_contacted_nhs", "think_had_covid_admitted_to_hospital", "other_covid_infection_test"]:
+        df = df.withColumn(
+            col,
+            F.when(
+                (~F.col("think_had_covid").eqNullSafe("Yes"))
+                & (F.col("other_covid_infection_test_results").isNull())
+                & (F.col("survey_response_dataset_major_version") == 0)
+                & (
+                    reduce(add, [F.when(F.col(c).isin(["Yes", "Positive"]), 1).otherwise(0) for c in covid_test_cols])
```

Comment:
The result for other_covid_infection_test are mapped to: 
[0] Any tests negative, but none positive; **[1] One or more positive test(s);** [2] Waiting for all results; [9] All Tests failed
So depending on whether this comes after the mapping this might need to be changed to the bold string or '1'.